### PR TITLE
Add in non-root user, docker secret support, and more

### LINKED ISSUE #6: Upgrades
Hello,

I used your repository as a starting point for creating my own flavor with the same ability. My version has several improvements that I wanted to offer you. There may be more that I'm missing but I think these are several very useful inclusions.

1. Running as the non root user (more secure)
2. Runs within the same container as the active database (fewer running containers)
3. The ability to define a different dump directory. 
4. Loading docker secrets for the password

If you're interested I can include them in a pull request.

### PR DESCRIPTION
**Pull request by gauntl3t12**
Closes: https://github.com/Martlark/pg_dump/issues/6

Adds:
- Non root user
- Fix deprecated label
- Runs in 1 container instead of 2
- README updates
- docker secrets support
- Sample docker-compose yml
- Env name updates to align more with PSQL
- Log env vars option for debugging

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -2,25 +2,65 @@
...
+then
+   COMMAND=${1:-dump-cron}
+fi
+
 CRON_SCHEDULE=${CRON_SCHEDULE:-0 1 * * *}
 PREFIX=${PREFIX:-dump}
 PGUSER=${PGUSER:-postgres}
-PGDB=${PGDB:-postgres}
-PGHOST=${PGHOST:-db}
+POSTGRES_DB=${POSTGRES_DB:-postgres}
```
Comment (Martlark) [inline entrypoint.sh]: I'd prefer to have some backwards compatibility by using PGDB as a default here

File: entrypoint.sh [hunk 2]
```diff
@@ -2,25 +2,65 @@
 # entrypoint.sh
 set -e
 
-COMMAND=${1:-dump-cron}
+if [[ -z $COMMAND ]];
```
Comment (Martlark) [inline entrypoint.sh:5]: This just repeats my use of default environment variables I think

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
+     "entrypoint.sh", \
+     "/" ]
 
-VOLUME /dump
+USER root
+
+VOLUME [ "/dump", "/data" ]
+
+RUN \
+apt-get update && \
```
Comment (Martlark) [inline Dockerfile]: I prefer these to be separate RUN items as it makes Docker cache layering easier

File: README.md [hunk 4]
```diff
@@ -31,29 +44,33 @@ Example with docker-compose:
...
 database:
-  image: postgres:12.1
-  volumes:
-    - ./persistent/data:/var/lib/postgresql/data
-  environment:
-    - POSTGRES_PASSWORD=SumPassw0rdHere
-    - POSTGRES_DB=postgres
-  restart: unless-stopped
-
-postgres-backup:
```
Comment (Martlark) [inline README.md:42]: Can you make two example docker-compose.yml files, one with a single combined dump/pg and one with them separate?  I like your enterprise on having fewer services, but I prefer to keep my services simple and task focused.

File: entrypoint.sh [hunk 5]
```diff
@@ -2,25 +2,65 @@
...
+   echo "PGPORT: ${PGPORT}"
+   echo "PGDUMP: ${PGDUMP}"
+   echo "POSTGRES_PASSWORD_FILE: ${POSTGRES_PASSWORD_FILE}"
+fi
+
+if [[ -f /run/secrets/db_password ]];
+then
+   source /run/secrets/db_password
+else
+   echo "ERROR: No password file found!"
```
Comment (Martlark) [inline entrypoint.sh]: make this a WARNING, not ERROR

File: entrypoint.sh [hunk 6]
```diff
@@ -2,25 +2,65 @@
...
+        echo "LOGFIFO: ${LOGFIFO}"
+    fi
     if [[ ! -e "${LOGFIFO}" ]]; then
         mkfifo "${LOGFIFO}"
     fi
-    CRON_ENV="PREFIX='$PREFIX'\nPGUSER='${PGUSER}'\nPGDB='${PGDB}'\nPGHOST='${PGHOST}'\nPGPORT='${PGPORT}'"
-    if [[ -n "${PGPASSWORD}" ]]; then
-        CRON_ENV="$CRON_ENV\nPGPASSWORD='${PGPASSWORD}'"
+    CRON_ENV="PREFIX='$PREFIX'\nPGUSER='${PGUSER}'\nPOSTGRES_DB='${POSTGRES_DB}'\nPGHOST='${PGHOST}'\nPGPORT='${PGPORT}'\nPGDUMP='${PGDUMP}'"
+    if [[ -n "${POSTGRES_PASSWORD}" ]]; then
```
Comment (Martlark) [inline entrypoint.sh:86]: POSTGRES_PASSWORD should be an alias for PGPASSWORD, and listed in the documentation table

File: entrypoint.sh [hunk 7]
```diff
@@ -30,9 +70,9 @@ elif [[ "${COMMAND}" == 'dump-cron' ]]; then
     echo -e "$CRON_ENV\n$CRON_SCHEDULE /dump.sh > $LOGFIFO 2>&1" | crontab -
     # crontab -l
     cron
-    tail -f "$LOGFIFO"
+    /usr/local/bin/docker-entrypoint.sh postgres
 else
-    echo "Unknown command $COMMAND"
+    echo "Unknown command: $COMMAND"
```
Comment (Martlark) [inline entrypoint.sh:105]: nice

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+version: "3.9"
+services:
+   database:
+     image: martlark/pg_dump:12.1
```
Comment (Martlark) [inline docker-compose.yml:4]: probably should make this a more recent version of postgres then I'm lazily using at home.

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+     container_name: postgres-backup
+     volumes:
+       - postgres-data:/data:rw
+       - postgres-backup:/dump:rw
+     environment:
+       - PGDATA=/data   # Where the SQL DB will store itself & backups will dump
+       - RETAIN_COUNT=1 # Keep this number of backups
+       - PGDB=postgres  # The name of the database to dump
```
Comment (Martlark) [inline docker-compose.yml]: this should be POSTGRES_DB I think to fit in with your new standard.

**Review by Martlark — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (gauntl3t12) [reply Dockerfile]: Fair, I am more accustomed to environments that look to reduce layers.

File: docker-compose.yml (hunk 8 above)
Comment (gauntl3t12) [reply docker-compose.yml:4]: I'll leave this to you. There's reasons to keep older versions.

File: docker-compose.yml (hunk 9 above)
Comment (gauntl3t12) [reply docker-compose.yml]: Good catch

File: entrypoint.sh (hunk 2 above)
Comment (gauntl3t12) [reply entrypoint.sh:5]: The issue is that you are using "1" not "COMMAND" this means that it does not work with the docker-compose. I was trying to maintain backwards compatibility here.

File: entrypoint.sh (hunk 5 above)
Comment (gauntl3t12) [reply entrypoint.sh]: Fixed with the aliasing taking place up here.

File: entrypoint.sh (hunk 6 above)
Comment (gauntl3t12) [reply entrypoint.sh:86]: See updates above.

File: README.md (hunk 4 above)
Comment (gauntl3t12) [reply README.md:42]: Supporting the single container and two container idea, required an update to the entrypoint.

Comment (gauntl3t12) [reply README.md:42]: I've created a second file named `docker-compose-double.yml` and included the contents in the README

**Review by gauntl3t12 — COMMENTED** [review]
Pushing the changes now for re-review @Martlark

File: entrypoint.sh (hunk 1 above)
Comment (gauntl3t12) [reply entrypoint.sh]: Added this

File: docker-compose.yml (hunk 8 above)
Comment (Martlark) [reply docker-compose.yml:4]: I'm ashamed to have not upgraded!

Comment (gauntl3t12) [reply docker-compose.yml:4]: Did you want me to upgrade this? Not sure what else you might be looking for prior to the merge

**Comment by Martlark** [comment]
Sorry, got a bit tied up in life.  I’ll review this weekend.

Regards,

Andrew R.

**Pull request merged**